### PR TITLE
refs #41: GitHub integration

### PR DESCRIPTION
**Pull request by edodusi**
Initial generic implementation of GitHub php client https://github.com/KnpLabs/php-github-api
Lists all _open_ issues for the given project (configurable in yaml).

Usage: `php spark.php github:search`

### CODE REVIEW

File: src/Sparkfabrik/Command/Github/GithubSearchCommand.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            'title' => 'Title',
+            'url' => 'URL',
+            'state' => 'State'
+            );
+            
+            // Render issue table.
+            $this->tableGithubOutput($output, $fields, $res, 'issues');
+            
+        } catch (Exception $e) {
+            return $output->writeln('<error>'. $e->getMessage() . '</error>');
```
Comment (paolomainardi) [inline src/Sparkfabrik/Command/Github/GithubSearchCommand.php:93]: exceptions are catched by symfony console itself, we should catch it just when we want add more useful informations, can you check it ?

Comment (paolomainardi) [inline src/Sparkfabrik/Command/Github/GithubSearchCommand.php:93]: please write an @expectedexception test

**Comment by paolomainardi** [comment]
@edodusi this MR cannot be merged are you still interested in working on it ?

**Comment by edodusi** [comment]
@paolomainardi yes please leave it open, I'll check it soon as possible

File: tests/SparkFabrik/Command/Github/GithubSearchCommandTest.php [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        );
+        $this->tester->execute($options);
+        $res = trim($this->tester->getDisplay());
+        $this->assertEquals('No issues found.', $res);
+    }
+
+    /**
+     * Test status syntax error.
+     *
+     * @group searchByStatus
```
Comment (paolomainardi) [inline tests/SparkFabrik/Command/Github/GithubSearchCommandTest.php]: use `@group github`

File: tests/SparkFabrik/Command/Github/GithubSearchCommandTest.php [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+            ->will($this->returnValue($this->githubClient));
+
+        // Set the mocked client.
+        $this->command->setService($this->service);
+    }
+
+    /**
+     * Test no issues found.
+     *
+     * @group noissues
```
Comment (paolomainardi) [inline tests/SparkFabrik/Command/Github/GithubSearchCommandTest.php]: use `@group github`

**Comment by edodusi** [comment]
:tada: :fireworks:  :smile:

**Pull request merged**